src/s1reader/s1_reader.py

```diff
@@ -569,18 +569,17 @@ def burst_from_xml(annotation_path: str, orbit_path: str, tiff_path: str,
         tree_lads = ET.parse(f_lads)
         product_annotation = ProductAnnotation.from_et(tree_lads)
 
+        rfi_annotation_path = annotation_path.replace('annotation/', 'annotation/rfi/rfi-')
         # Load RFI information if available
-        if ipf_version >= RFI_INFO_AVAILABLE_FROM:
-            rfi_annotation_path =\
-                    annotation_path.replace('annotation/',
-                                            'annotation/rfi/rfi-')
+        if os.path.exists(rfi_annotation_path) and ipf_version >= RFI_INFO_AVAILABLE_FROM:
```

Comment:
Considering that you are verifying the existence of the file, maybe you can remove `ipf_version >= RFI_INFO_AVAILABLE_FROM`? The only exception would be if there was some sort of RFI annotation prior to `RFI_INFO_AVAILABLE_FROM` with a different format

Reply:
Good point, changed. I think in earlier products, the "product structure" being different just means the file doesn't exist.
![image](https://github.com/opera-adt/s1-reader/assets/8291800/52c88ff6-0e69-4fe6-b93a-f07fba7b6a66)
https://sentinel.esa.int/documents/247904/1653442/DI-MPC-OTH-0540-1-0-RFI-Tech-Note.pdf/4b4fa95d-039f-5c78-fb90-06d307b3c13a?t=1644988601315

Reply:
Awesome. Thank you, @scottstanie , for checking and updating the code